### PR TITLE
[Feature](json_functions) support json-keys 

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
support json_keys 
sql reference MYSQL: https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html#function_json-keys
Issue Number: close #xxx

### CODE REVIEW

**Review by xiaokang — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/JsonKeys.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ * Returns NULL if any argument is NULL, the json_doc argument is not an object,
+ * or path, if given, does not locate an object.
+ * An error occurs if the json_doc argument is not a valid JSON document
+ * or the path argument is not a valid path expression.
+ */
+public class JsonKeys extends ScalarFunction
+        implements ExplicitlyCastableSignature, AlwaysNullable {
+
+    public static final List<FunctionSignature> SIGNATURES = ImmutableList.of(
+            FunctionSignature.ret(ArrayType.of(StringType.INSTANCE, false))
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/JsonKeys.java:47]: why `ArrayType.of(StringType.INSTANCE, false)` not `ArrayType.of(StringType.INSTANCE)`. we support array with contains null = false?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/BuiltinScalarFunctions.java [hunk 2]
```diff
@@ -713,6 +714,7 @@ public class BuiltinScalarFunctions implements FunctionHelper {
             scalar(JsonbType.class, "jsonb_type"),
             scalar(JsonLength.class, "json_length"),
             scalar(JsonContains.class, "json_contains"),
+            scalar(JsonKeys.class, "json_keys"),
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/BuiltinScalarFunctions.java:717]: legacy planner has a alias name 'jsonb_keys', why nereids not add it?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/JsonKeys.java (hunk 1 above)
Comment (amorynan) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/JsonKeys.java:47]: no we do not support ,  I would erase `false`

**Comment by amorynan** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/BuiltinScalarFunctions.java [hunk 3]
```diff
@@ -713,6 +714,8 @@ public class BuiltinScalarFunctions implements FunctionHelper {
...
+            scalar(JsonKeys.class, "jsonb_keys"),
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/BuiltinScalarFunctions.java:718]:
```suggestion
            scalar(JsonKeys.class, "json_keys", "jsonb_keys"),
```

File: regression-test/data/jsonb_p0/test_jsonb_load_and_function.out
Comment (morrySnow) [inline regression-test/data/jsonb_p0/test_jsonb_load_and_function.out:1]: add case in regression-test/suites/nereids_function_p0/scalar_function/J.groovy

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/functions/function_jsonb.cpp [hunk 4]
```diff
@@ -431,6 +431,176 @@ class FunctionJsonbExtract : public IFunction {
...
+            data_null_map = &nullable->get_null_map_data();
+        }
+        const ColumnString* col_from_string = check_and_get_column<ColumnString>(jsonb_data_column);
+
+        // prepare parse path column prepare, maybe we do not have path column
+        ColumnPtr jsonb_path_column = nullptr;
+        const ColumnString* jsonb_path_col = nullptr;
+        bool path_const = false;
+        const NullMap* path_null_map = nullptr;
+        if (arguments.size() == 2) {
```
Comment (eldenmoon) [inline be/src/vec/functions/function_jsonb.cpp:467]: put check here
```
if (arguments.size() != 1 and arguments.size() != 2) {
            // here has argument param error
            return Status::InvalidArgument("json_keys should have 1 or 2 arguments");
        }
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/JsonKeys.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```
Comment (eldenmoon) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/JsonKeys.java:1]: add document to doris-website

**Comment by amorynan** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — APPROVED** [review]
lgtm

**Review by morrySnow — APPROVED** [review]

**Pull request merged**